src/api/pool.go

```diff
@@ -9,18 +9,40 @@ import (
 )
 
 type Pool struct {
-	client          *Client         // http client
+	client          *Client         // resty client
 	ctx             context.Context // context of the parallel work
 	workers         *errgroup.Group // error group -> if one worker fails, all will be stopped
 	counter         sync.WaitGroup  // detect when all requests are processed
 	sendersCount    int
 	processorsCount int
-	processor       func(pool *Pool, response *resty.Response) error
+	processor       func(pool *Pool, response *PoolResponse) error
+	done            chan struct{}
 	requests        chan *resty.Request
-	responses       chan *resty.Response
+	responses       chan *PoolResponse
```

Comment:
Su tu nejake zmeny aj v `pool`. Povodne sietove chyby zastavili vykonavanie requestov ... a http chyby sa spracovavali v processore.... prerobil som to tak, ze processor rozhoduje o vsetkych chyba ... vsetky zmeny su ohladom toho.